include/lbann/data_coordinator/data_coordinator.hpp

```diff
@@ -281,7 +281,8 @@ class data_coordinator
 
   bool at_new_epoch() const;
 
-  virtual void register_active_data_field(data_field_type const data_field);
+  virtual void register_active_data_field(data_field_type const data_field,
+                                          std::vector<El::Int> const data_field_dim_map);
```

Comment:
Do you mean
```
  virtual void register_active_data_field(data_field_type const& data_field,
                                          std::vector<El::Int> const& data_field_dim_map);
```
? Generally, C++ does not declare "pass-by-value" parameters to be `const`. I would suggest either removing the `const`s or making these "pass-by-const-reference" parameters.